js/ui/appDisplay.js

```diff
@@ -754,6 +753,17 @@ const AllView = new Lang.Class({
         this.parent();
     },
 
+    getLayoutIds: function() {
+        let layoutIds = this.parent();
+
+        // AllView also has the App Center icon appended at the end of the list.
+        // For Drag n' Drop work correctly, we must take this icon into account
+        // when calculating the diff between before and after the DnD.
+        layoutIds.push(EOS_APP_CENTER_ID);
```

Comment:
This should be conditional to the `enable-app-center` GSetting key at least. See `_maybeAddAppCenterIcon()`